### ISSUE TITLE
Project platform settings other than "Any CPU" are ignored

### ISSUE THREAD

**Issue by bashtavenko** [dotnet/roslyn#6535]
If project in the solution has platform settings other than "Any CPU" (x86, x64), Roslyn picks incorrect configuration. For example, if one of the projects has OutputPath of "bin\x86\Debug"

```
 <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|x86'">
    <DebugSymbols>true</DebugSymbols>
    <OutputPath>bin\x86\Debug\</OutputPath>
    <DefineConstants>DEBUG;TRACE</DefineConstants>
    <DebugType>full</DebugType>
    <PlatformTarget>x86</PlatformTarget>
```

This code returns "bin\Debug" in project,OutputFilePath. Other project properties are also incorrect and correspond to "Any CPU" as opposed to "x86".

```
using (var workspace = MSBuildWorkspace.Create())
{
    var solution = workspace.OpenSolutionAsync(solutionPath).Result;
    foreach (var project in solution.Projects)
    {
        var projectModulePath = project.OutputFilePath;
        //...
    }
}
```

**Comment by Pilchie** [comment]
You should be able to pass `Configuration` and `Platform` as part of the `properties` parameter of [`MSBuildWorkspace.Create(IDictionary<string, string> properties)`](http://source.roslyn.io/Microsoft.CodeAnalysis.Workspaces.Desktop/R/02c87df4b2a1da1a.html)

**Comment by StanBPublic** [comment]
When there are multiple configurations in the solution, Visual Studio sets  platform drop down to "Mixed Platforms" automatically

```
GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|Any CPU = Debug|Any CPU
		Debug|Mixed Platforms = Debug|Mixed Platforms
		Debug|x86 = Debug|x86
	EndGlobalSection
```
.. and thus after build the assemblies go to ".bin\x86\Debug.." folder. When the same solution is opened in Roslyn, it appears that it picks up "Debug|Any CPU" configuration and thinks that assemblies should be in "bin\Debug", but they aren't there.

I understand that I can pass Configuration and Platform, but given a solution, how do I know what these parameters should be?

**Comment by Pilchie** [comment]
Oh, good point.  I didn't really think about the default configurations stored in the .sln file.  We should probably use those when loading an sln.  I'm not sure if MSBuild added sufficient solution parsing APIs for that, but if so, we'd want to look at updating the code around [here](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces.Desktop/Workspace/MSBuild/MSBuildProjectLoader.cs,142)

**Comment by Pilchie** [comment]
Is #6709 the same issue?

**Comment by mattwar** [comment]
Yes. Msbuild workspace has no logic for picking a configuration. Currently it requires the caller to set the configuration as a property. If not specified it just defaults to whatever the default of the project file.

**Comment by Ziriax** [comment]
Today we upgraded to Visual Studio 2017 15.3.2. One of our projects uses Roslyn's MSBuildWorkspace (version 2.3.1), and we're explicitly passing x86 as the platform and Diagnostics as our (custom) configuration. This worked fine until today, but after the upgrade each Project is loaded completely empty (no document, empty OutputPath, etc). On another workstation where we did not do the upgrade, everything still works fine. I tried doing a system restore but that completely broke Visual Studio. Any idea what could be going wrong? We have an MSDN Enterprise subscription, I'm waiting for support from that source too.
